### PR TITLE
adds xarray return option to _extract_bin_files

### PR DESCRIPTION
**Pull request by ackagel**
closes issue #20 

direct ndarray return would've been a little awkward with 5 dimensions (fov, pulse/intensity/area, x, y, channel), so I settled for an xarray return.

### CODE REVIEW

**Comment by ackagel** [comment]
@ngreenwald lmk if you want me to expand on the testing at all.  for now, I'm just making sure that the correct dimension names are returned.  If we want to check for full correctness (as in file writes match direct returns) then I can move those calls into the tempfile context and check in there.  That felt a little overkill, but I can totally implement that if you think it's worth it.

File: tests/bin_files_test.py [hunk 1]
```diff
@@ -203,6 +203,10 @@ def test_extract_bin_files(test_dir, fov, panel, intensities, filepath_checks):
         bin_files.extract_bin_files(test_dir, tmpdir, None, panel, intensities, time_res)
         filepath_checks(tmpdir, fov['json'].split('.')[0], panel['Target'].values, intensities)
 
+    # test xr write out
+    test_xr = bin_files.extract_bin_files(test_dir, None, None, panel, intensities, time_res)
+    assert(list(test_xr.dims) == ['fov', 'type', 'x', 'y', 'channel'])
```
Comment (ngreenwald) [inline tests/bin_files_test.py:208]: I don't think an exact correctness test is necessary, but adding a check that the number of FOVs is what's expected and that the different FOVs aren't identical should be enough to ensure that the concat logic is working

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Just a small addition. Can you also update the make panels function, either in here or a separate PR?

File: tests/bin_files_test.py (hunk 1 above)
Comment (ackagel) [reply tests/bin_files_test.py:208]: right now, I'm only testing `extract_bin_files` on one fov (one moly and one tissue).  If you have another small tissue or moly fov from the same run, I could add that to the testing data to cover that case.  I could also just merge copies of the tissue+moly into one run, but I'm not 100% sure they'll play nice, before trying it.

Comment (ngreenwald) [reply tests/bin_files_test.py:208]: Yeah, the tiny_bin_files folder I sent you has three FOVs in it

**Review by ngreenwald — APPROVED** [review]
Looks good

**Pull request merged**